src/internal/m365/collection/exchange/contacts_backup_test.go

```diff
@@ -0,0 +1,31 @@
+package exchange
+
+import (
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/suite"
+
+	"github.com/alcionai/corso/src/internal/tester"
+	"github.com/alcionai/corso/src/pkg/control"
+	"github.com/alcionai/corso/src/pkg/fault"
+	"github.com/alcionai/corso/src/pkg/services/m365/api"
+)
+
+type ContactsBackupHandlerUnitSuite struct {
+	tester.Suite
+}
+
+func TestContactsBackupHandlerUnitSuite(t *testing.T) {
+	suite.Run(t, &ContactsBackupHandlerUnitSuite{Suite: tester.NewUnitSuite(t)})
+}
+
+func (suite *ContactsBackupHandlerUnitSuite) TestHandler_CanSkipItemFailure() {
+	t := suite.T()
+
+	h := newContactBackupHandler(api.Client{})
+	cause, result := h.CanSkipItemFailure(nil, "", "", control.Options{})
```

Comment:
The tests for events aren't generically applicable for mail and contacts.  Primarily because the configuration that the tests depend on is also targeted to calendar events.  I could write in nil and non-nil checks with arbitrary data, but that seems like moot effort given the flat return of the func.  If anything, I kinda prefer the singular test, as evidence that we aren't considering any difference in inputs for those handlers; the output is always, simply, false.

Reply:
> The tests for events aren't generically applicable for mail and contacts

The new functions you added are still going to be called during backup because the handler is passed in as the skip checker in later PRs. That makes their implementation just as important as the implementation of the events handler. Part of writing code and tests is making sure the _wrong_ behavior doesn't happen. It can sometimes be just as much of the work as making sure the _right_ behavior does happen

This branch [dnm-bug-1](https://github.com/alcionai/corso/tree/dnm-bug-1) shows that unit tests will pass for contacts and mail _even if the implementation will lead to bugs in an actual backup_. What is being tested in this PR is "the implementation works right conditioned on the fact that the implementation is right" not that the implementation is correct no matter what it is

Reply:
To be fair, I think you're pointing out the ease of producing these conditions.  Not that it's possible.  The statement of possibility is a strawman argument, because nearly any function can be manufactured to introduce bugs while satisfying tests.  But that was particularly easy to do, I'll give you that.

Reply:
I think we may have different views of what tests are supposed to represent. In my mind tests should not have any knowledge of the implementation unless they're checking some optimizations. They should be blackbox tests that check for a given set of inputs the correct outputs for the component in question are returned. This requires having a spec of the expected behavior. In the case of mail and contacts, the spec should be "always return false when `CanSkipItemFailure` is called". However, a test with a single test case is not enough to show that for all inputs it returns false. Now, we can't really test "all inputs", so instead we choose representative sets of inputs, like the ones for the events test

Having tests that encode the spec expectations also helps guard against regressions in the future. The example of a bug I gave is a simple one that could be caught in review, if it's not overlooked because it's grouped together with a lot of other changes. There's a lot of other bugs that are equally dangerous that are much less obvious because they deal with subtle interactions between things, [like an error message changing](https://github.com/alcionai/corso/blob/5e8407a9702368b80c6d856b7ba309561a38e445/src/pkg/services/m365/api/mail.go#L588) or something. If these reside on code executed for edge cases and there's no tests for it then it could take a really long time to catch these issues

Reply:
It's true we differ on these things.  Admittedly, though, i could probably shift a bit further adapting for that viewpoint and be better in life as a result.